content/en/blog/_posts/2022-03-31-Kubernetes-1-24-deprecations-and-removals

```diff
@@ -0,0 +1,85 @@
+    
+## The Kubernetes API removal process
+
+Kubernetes contains a large number of components that evolve over time. In some cases, this evolution results in APIs, flags, or entire features, being removed. To prevent users from facing breaking changes, Kubernetes adopted a feature [deprecation policy](/docs/reference/using-api/deprecation-policy/). This policy ensures that stable APIs may only be deprecated when a newer stable version of that same API is available and that APIs have a minimum lifetime depending on their stability level: 
+
+* Generally available (GA) or stable API versions may be marked as deprecated but must not be removed within a major version of Kubernetes. 
+* Beta or pre-release API versions must be supported for 3 releases after deprecation. 
+* Alpha or experimental API versions may be removed in any release without prior deprecation notice. 
+
+Kubernetes will continue to remove APIs by following the deprecation policy regardless of whether an API is removed due to a beta feature graduating to stable or because that API was not proven to be successful and will also make sure migration options are documented.
+
+    
+### Looking ahead
+
+The Kubernetes 1.25 and 1.26 releases planned for later this year will stop serving beta versions of several currently stable Kubernetes APIs. The 1.25 release will also remove PodSecurityPolicy, which was deprecated with Kubernetes 1.21 and will not graduate to stable. See [PodSecurityPolicy Deprecation: Past, Present, and Future](/blog/2021/04/06/podsecuritypolicy-deprecation-past-present-and-future/) for more information.
+
+The official [list of API removals planned for Kubernetes 1.25](/docs/reference/using-api/deprecation-guide/#v1-25) is:
+
+* The beta CronJob API (batch/v1beta1)
+* The beta EndpointSlice API (discovery.k8s.io/v1beta1)
+* The beta Event API (events.k8s.io/v1beta1)
+* The beta HorizontalPodAutoscaler API (autoscaling/v2beta1)
+* The beta PodDisruptionBudget API (policy/v1beta1)
+* The beta PodSecurityPolicy API (policy/v1beta1)
+* The beta RuntimeClass API (node.k8s.io/v1beta1)
+
+
+The official [list of API removals planned for Kubernetes 1.26](/docs/reference/using-api/deprecation-guide/#v1-26) is:
+
+* The beta FlowSchema and PriorityLevelConfiguration APIs (flowcontrol.apiserver.k8s.io/v1beta1)
+* The beta HorizontalPodAutoscaler API (autoscaling/v2beta2)
+
+    
+## Want to know more?
```

Comment:
I'd mention dockershim in this section too.

Reply:
I didn't see Dockershim called out in the https://kubernetes.io/docs/reference/using-api/deprecation-guide/. What's the best way to verify which version it will be removed?

Reply:
https://github.com/kubernetes/enhancements/blob/master/keps/sig-node/2221-remove-dockershim/kep.yaml details project _intent_ and also makes it clear that SIG Node (well, and SIG Release, the gatekeepers of all that we release) are the folks to ask.

Reply:
We're definitely removing the dockershim in v.124, mind.